ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -167,24 +201,18 @@ public void execute(
           value
       );
 
-      producer.sendRecord(record, serviceContext, config.getProducerClientConfigProps());
-    } catch (final TopicAuthorizationException e) {
-      // TopicAuthorizationException does not give much detailed information about why it failed,
-      // except which topics are denied. Here we just add the ACL to make the error message
-      // consistent with other authorization error messages.
-      final Exception rootCause = new KsqlTopicAuthorizationException(
-          AclOperation.WRITE,
-          e.unauthorizedTopics()
-      );
-
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), rootCause);
-    } catch (final Exception e) {
-      throw new KsqlException("Failed to insert values into stream/table: "
-          + insertValues.getTarget().name(), e);
+      return record;
+    } catch (Exception e) {
+      throw new KsqlStatementException(
+          createInsertFailedExceptionMessage(tableStreamName) + " " + e.getMessage(),
+          statement.getStatementText(), e);
     }
   }
 
+  private String createInsertFailedExceptionMessage(final String streamTableName) {
+    return "Failed to insert values into stream/table: " + streamTableName + ".";
```

Comment:
nit: personally, I'd drop the `stream/table` - looks whack and is superfluous IMHO. I'd also wrap the name in quotes

```suggestion
    return "Failed to insert values into '" + streamTableName + "'.";
```

Reply:
I think this was how it used to be ;)
I added it back in after your comment yesterday about not changing KsqlTestingToolTest. The original version of which asserts the stream/table name.

Reply:
Ah, you misunderstood. My bad. Should have explained better.  Make this suggested change, I'll comment below about what I meant about keeping the test as is.